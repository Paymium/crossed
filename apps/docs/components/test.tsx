import { ButtonDemo } from '@crossed/demo';

export default ButtonDemo;
