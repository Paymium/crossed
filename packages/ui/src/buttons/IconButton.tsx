/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { Pressable, PressableProps } from 'react-native';
import {
  composeStyles,
  inlineStyle,
  CrossedMethods,
  useTheme,
  Registry,
} from '@crossed/styled';
import { cloneElement, isValidElement, memo, useCallback } from 'react';

export interface IconButtonProps extends Omit<PressableProps, 'style'> {
  /**
   * Crossed style
   */
  style?: CrossedMethods<any>;
}

export const iconButtonStyles = inlineStyle(({ components: { Action } }) => ({
  'base': {
    borderRadius: 50,
    backgroundColor: Action.icon.default.background,
    borderColor: Action.icon.default.border,
    height: 40,
    width: 40,
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 1,
  },
  ':hover': {
    backgroundColor: Action.icon.hover.background,
    borderColor: Action.icon.hover.border,
  },
  ':active': {
    backgroundColor: Action.icon.active.background,
    borderColor: Action.icon.active.border,
    outlineColor: Action.icon.focus.border,
  },
  ':disabled': { opacity: 0.5 },
  'web': {
    ':focus-visible': {
      backgroundColor: Action.icon.focus.background,
      outlineColor: Action.icon.focus.border,
    },
  },
}));

export const IconButton = memo<IconButtonProps>(
  ({ children, style, ...props }) => {
    const { components } = useTheme();
    const getStyle = useCallback(
      ({
        hovered,
        pressed,
        focused,
      }: {
        hovered?: boolean;
        pressed?: boolean;
        focused?: boolean;
      }) => {
        return composeStyles(iconButtonStyles, style).style({
          hover: hovered,
          focus: focused,
          active: pressed,
          disabled: !!props.disabled,
        }).style;
      },
      [props.disabled, Registry.themeName, style]
    );
    const child = useCallback(
      (e: any) => {
        const tmpChild =
          typeof children === 'function' ? children(e) : children;

        if (isValidElement(tmpChild)) {
          return cloneElement(tmpChild, {
            size: 16,
            color: components.Action.icon.default.color,
          } as any);
        }
        return tmpChild;
      },
      [children]
    );
    return (
      <Pressable role="button" {...props} style={getStyle}>
        {child}
      </Pressable>
    );
  }
);
