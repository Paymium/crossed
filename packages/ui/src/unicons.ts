declare module '@iconscout/react-native-unicons' {
  import type { ComponentType } from 'react';

  type IconComponent = ComponentType<{
    size?: number | string;
    color?: 'string';
  }>;

  export const UilMessage: IconComponent;
}
