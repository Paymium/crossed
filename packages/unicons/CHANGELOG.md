# @crossed/unicons

## 1.0.0-beta.13

### Patch Changes

- Updated dependencies [ffc415d]
- Updated dependencies [ffc415d]
  - @crossed/styled@0.14.0-beta.10
  - @crossed/theme@1.0.1-beta.3

## 1.0.0-beta.12

### Patch Changes

- Updated dependencies [20288fe]
- Updated dependencies [20288fe]
  - @crossed/styled@0.14.0-beta.9
  - @crossed/theme@1.0.1-beta.2

## 1.0.0-beta.11

### Patch Changes

- 73aac8d: remove theme

## 1.0.0-beta.10

### Patch Changes

- 063b0b6: remove theme and force build

## 1.0.0-beta.8

### Patch Changes

- Updated dependencies [4f81cb4]
  - @crossed/styled@0.14.0-beta.8
  - @crossed/theme@1.0.1-beta.1

## 1.0.0-beta.7

### Minor Changes

- 4ae3b66: add CheckCircle and X

### Patch Changes

- 0cf4ddc: add select, checkbox, radio
- Updated dependencies [c1e88c5]
- Updated dependencies [3a58e76]
- Updated dependencies [0cf4ddc]
- Updated dependencies [511e2e5]
- Updated dependencies [017fd84]
- Updated dependencies [511e2e5]
- Updated dependencies [f1c8cbd]
  - @crossed/styled@0.14.0-beta.7
  - @crossed/core@0.8.1-beta.3
  - @crossed/theme@1.0.1-beta.0

## 1.0.0-beta.6

### Patch Changes

- Updated dependencies
  - @crossed/core@0.8.1-beta.2
  - @crossed/styled@0.14.0-beta.6

## 1.0.0-beta.5

### Patch Changes

- Updated dependencies [63171a9]
  - @crossed/styled@0.14.0-beta.5

## 1.0.0-beta.4

### Patch Changes

- Updated dependencies
  - @crossed/styled@0.14.0-beta.4
  - @crossed/core@0.8.1-beta.1

## 1.0.0-beta.3

### Patch Changes

- Updated dependencies [7a7a589]
  - @crossed/styled@0.14.0-beta.3

## 1.0.0-beta.2

### Patch Changes

- Updated dependencies
  - @crossed/core@0.8.1-beta.0
  - @crossed/styled@0.14.0-beta.2

## 1.0.0-beta.1

### Patch Changes

- Updated dependencies
  - @crossed/styled@0.14.0-beta.1

## 1.0.0-beta.0

### Patch Changes

- Updated dependencies [d9f9372]
- Updated dependencies [31e0003]
- Updated dependencies [c7e44dc]
- Updated dependencies [8917db4]
  - @crossed/styled@0.14.0-beta.0

## 0.6.0

### Minor Changes

- 70a832e: Breaking changes: remove :dark and :light key, remove postcss plugin

## 0.5.0

### Minor Changes

- some increase packages

## 0.4.0

### Minor Changes

- add extends

## 0.3.0

### Minor Changes

- remove animation

## 0.2.2

### Patch Changes

- fix build

## 0.2.1

### Patch Changes

- test

## 0.2.0

### Minor Changes

- 1d68c0c: First beta
