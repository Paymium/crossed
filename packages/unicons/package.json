{
  "name": "@crossed/unicons",
  "description": "unicons",
  "version": "0.6.0",
  "main": "lib/commonjs/index",
  "types": "lib/typescript/index.d.ts",
  "module": "lib/module/index",
  "react-native": "src/index",
  "source": "src/index",
  "typings": "lib/typescript/index.d.ts",
  "scripts": {
    "release": "release-it",
    "clean": "rm -rf lib",
    "watch": "crossed-build --watch --project-path ./tsconfig.build.json",
    "build": "crossed-build --project-path ./tsconfig.build.json",
    "dev:web": "cd example/native && yarn web --clear",
    "type-check": "tsc --noEmit"
  },
  "devDependencies": {
    "@crossed/build": "*",
    "@types/react": "^18.2.21",
    "@types/react-native": "^0.72.2",
    "babel-plugin-transform-remove-console": "^6.9.4",
    "jest": "^29.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-native": "^0.72.3",
    "react-native-svg": "^13.10.0",
    "react-native-web": "^0.19.7",
    "tsconfig": "*"
  },
  "peerDependencies": {
    "react-native-svg": "^13.10.0",
    "react": "^18.2.0"
  },
  "dependencies": {
    "@iconscout/react-native-unicons": "^2.0.2"
  },
  "files": [
    "lib/",
    "src/"
  ]
}
