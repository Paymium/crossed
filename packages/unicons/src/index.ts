/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import './types';
export * from './ChevronRight';
export * from './ChevronDown';
export * from './ChevronUp';
export * from './CheckCircle';
export * from './Github';
export * from './Moon';
export * from './MousePointerClick';
export * from './Sun';
export * from './Check';
export * from './X';
