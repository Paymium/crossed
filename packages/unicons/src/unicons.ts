export * from '@iconscout/react-native-unicons';
