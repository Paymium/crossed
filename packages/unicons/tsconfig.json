{
  "include": ["./src", "src/unicons.ts"],
  "exclude": ["node_modules", "example"],
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./lib/typescript",
    "ignoreDeprecations": "5.0",
    "noEmit": true,
    "declaration": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": true,
    "esModuleInterop": true,
    "importsNotUsedAsValues": "error",
    "forceConsistentCasingInFileNames": true,
    "jsx": "react-jsx",
    "lib": ["esnext", "dom"],
    "module": "esnext",
    "moduleResolution": "node",
    "noFallthroughCasesInSwitch": true,
    "noImplicitReturns": true,
    "noImplicitUseStrict": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true,
    "target": "esnext",
  }
}
