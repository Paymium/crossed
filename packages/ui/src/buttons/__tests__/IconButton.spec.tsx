/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import React, { act } from 'react';
import { IconButton } from '../IconButton';
import { X } from '@crossed/unicons';
import { userEvent, render, screen } from '@crossed/test';

describe('IconButton', () => {
  it('renders correctly', () => {
    render(<IconButton />);
    const button = screen.getByRole('button');
    expect(button).toBeTruthy();
  });

  it('renders the provided children', () => {
    render(
      <IconButton>
        <X />
      </IconButton>
    );

    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('calls onPress when pressed', async () => {
    const onPressMock = jest.fn();
    render(<IconButton onPress={onPressMock} />);
    const button = screen.getByRole('button');

    await act(() => userEvent.click(button));
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('does not call onPress when disabled', async () => {
    const onPressMock = jest.fn();
    render(<IconButton onPress={onPressMock} disabled />);
    const button = screen.getByRole('button');

    await act(() => userEvent.click(button, { pointerEventsCheck: 0 }));
    expect(onPressMock).not.toHaveBeenCalled();
  });

  it('renders function-based children correctly', () => {
    const childMock = jest.fn(() => <X />);
    render(<IconButton>{childMock}</IconButton>);

    // Vérifie que la fonction enfant est appelée
    expect(childMock).toHaveBeenCalled();

    // Vérifie que l'icône est rendue
    expect(screen.getByRole('button')).toMatchSnapshot();
  });
});
