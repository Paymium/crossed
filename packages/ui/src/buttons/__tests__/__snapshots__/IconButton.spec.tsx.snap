// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders function-based children correctly 1`] = `
<button
  class="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
  role="button"
  style="border-top-left-radius: 50px; border-top-right-radius: 50px; border-bottom-right-radius: 50px; border-bottom-left-radius: 50px; height: 40px; width: 40px; align-items: center; justify-content: center; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px;"
  tabindex="0"
  type="button"
>
  <svg
    fill="none"
    height="16"
    stroke="var(--components--action-icon-default-color)"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="16"
  >
    <path
      d="M18 6 6 18"
      stroke="var(--components--action-icon-default-color)"
    />
    <path
      d="m6 6 12 12"
      stroke="var(--components--action-icon-default-color)"
    />
  </svg>
</button>
`;

exports[`IconButton renders the provided children 1`] = `
<button
  class="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
  role="button"
  style="border-top-left-radius: 50px; border-top-right-radius: 50px; border-bottom-right-radius: 50px; border-bottom-left-radius: 50px; height: 40px; width: 40px; align-items: center; justify-content: center; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px;"
  tabindex="0"
  type="button"
>
  <svg
    fill="none"
    height="16"
    stroke="var(--components--action-icon-default-color)"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="16"
  >
    <path
      d="M18 6 6 18"
      stroke="var(--components--action-icon-default-color)"
    />
    <path
      d="m6 6 12 12"
      stroke="var(--components--action-icon-default-color)"
    />
  </svg>
</button>
`;
